test/ctia/http/generative/fulltext_search_test.clj

```diff
@@ -0,0 +1,223 @@
+(ns ctia.http.generative.fulltext-search-test
+  (:require
+   [clojure.test :refer [deftest testing is use-fixtures join-fixtures]]
+   [clojure.test.check.generators :as gen]
+   [ctia.auth.capabilities :as capabilities]
+   [ctia.auth.threatgrid :refer [map->Identity]]
+   [ctia.bundle.core :as bundle]
+   [ctia.entity.entities :as entities]
+   [ctia.http.generative.properties :as prop]
+   [ctia.lib.utils :as utils]
+   [ctia.test-helpers.core :as helpers]
+   [ctia.test-helpers.es :as es-helpers]
+   [ctia.test-helpers.fake-whoami-service :as whoami-helpers]
+   [ctia.test-helpers.search :as th.search]
+   [ctim.examples.bundles :refer [bundle-maximal]]
+   [ctim.schemas.bundle :as bundle.schema]
+   [ductile.index :as es-index]
+   [puppetlabs.trapperkeeper.app :as app]))
+
+(def ^:private login
+  (map->Identity {:login  "foouser"
+                  :groups ["foogroup"]}))
+
+(use-fixtures :once (join-fixtures [es-helpers/fixture-properties:es-store
+                                    whoami-helpers/fixture-server]))
+
+(def bundle-entity-field-names
+  "extracted non-essential Bundle keys"
+  (->>
+   bundle.schema/objects-entries
+   (into bundle.schema/references-entries)
+   (mapcat #(-> % :key :values))
+   (set)))
+
+(defn- entities-gen
+  "Generator for a map where each k/v represents entity-type/list of entities (of
+  that type)"
+  [& entity-keys]
+  (let [gens (map
+              (fn [e]
+                (-> (->> e
+                         entities/entity-plural->entity
+                         ffirst
+                         name
+                         (str "max-new-")
+                         prop/spec-gen
+                         ;; remove IDs so it can be used it in Bundle import
+                         (gen/fmap #(dissoc % :id)))
+                    (gen/vector 5 11)))
+              entity-keys)]
+    (gen/bind
+     (apply gen/tuple gens)
+     (fn [entity-maps]
+       (gen/return
+        (zipmap entity-keys entity-maps))))))
+
+(defn- bundle-gen-for
```

Comment:
why do you need a generator?
From what I red, you only search on given fields for which you set a fix value, don't you?
Using CTIM minimal examples should be enough.

Reply:
The idea was to test the search with random values of different entities, i.e.:
 
- create a few random assets, incidents, etc. 
- change some fields in chosen entities (e.g.: set the Title for two incidents to be specific, while the rest would remain random values)
- issue a query like: "get me entities with specific title"
- ensure that only two relevant incidents come back (even though we have more)